src/makeInfinityAdder.js

```diff
@@ -6,7 +6,26 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  let prevState;
+
+  const adder = (num) => {
+    if (prevState === undefined) {
+      sum = 0;
+    }
+
+    if (num !== undefined) {
+      sum += num;
+      prevState = num;
+
+      return adder;
+    }
+    prevState = num;
```

Comment:
You could create here variable for example `const result = sum`
after that reset your `sum`  like `sum = 0` and return the `result` or do it the same inside if statement
Also would be nice to add some error handler for type of the num because if we add some string instead of number we will receieve the value we didn't expect